### PR TITLE
[cdc_rsync] Improve README

### PR DESCRIPTION
**Pull request by ljusten**
Adds more info about how cdc_rsync works and why it's faster.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,22 +38,93 @@ version of the files available in the target directory.
...
+We also ran the experiment with the native Linux `rsync`, i.e syncing Linux to
+Linux, to rule out issues with Cygwin. Linux `rsync` performed on average 35%
+worse than Cygwin `rsync`, which can be attributed to CPU differences. We did
+not include the graph due to this.
 
 <p align="center">
   <img src="docs/cdc_rsync_vs_cygwin_rsync.png" alt="Comparison of cdc_rsync and Linux rsync running in Cygwin" width="753" />
 </p>
 
+So how does it work and why is it faster? The **standard Linux `rsync`** splits
```
Comment (chrschng) [inline README.md:62]: Can you add a sub-heading here to structure the text better? Maybe `Implementation details` or `How it works`?

File: README.md [hunk 2]
```diff
@@ -38,22 +38,93 @@ version of the files available in the target directory.
...
+
+If the file is modified in the middle, e.g. by inserting `xxxx` after `567`,
+this usually means that <span style="color: red">the modified chunks as well as
+all subsequent chunks</span> change.
+
+<p align="center">
+  <img src="docs/fixed_size_chunks_inserted.png" alt="Fixed size chunks after inserting data" width="301" />
+</p>
+
+The **original `rsync` algorithm** hashes the chunks of the remote "old" file
```
Comment (chrschng) [inline README.md:77]: The bold font looks weird when rendered in conjunction with `rsync`, I don't think it helps with the readability of the text:
https://github.com/google/cdc-file-transfer/tree/docs#readme

Comment (chrschng) [inline README.md:77]: "original rsync algorithm" sounds as if there are further versions of it. How about "standard rsync algorithm"?

File: README.md [hunk 3]
```diff
@@ -38,22 +38,93 @@ version of the files available in the target directory.
...
+lookup, an `add` and an `and` operation for each input byte. This is cheaper
+than the hash map lookup for the original `rsync` algorithm.
+
+Because of this, the **`cdc_rsync` algorithm** is faster than the original
+`rsync`. It is also simpler. Since chunk boundaries move along with insertions
+or deletions, the task to match local and remote hashes is a trivial set
+difference operation. It does not involve a per-byte hash map lookup.
+
+<p align="center">
+  <img src="docs/cdc_rsync_animation.gif" alt="Syncing a file with cdc_rsync" width="857" />
```
Comment (chrschng) [inline README.md:124]: That's a cool animation!

**Review by chrschng — DISMISSED** [review]
This is nice, thanks!

File: README.md (hunk 1 above)
Comment (ljusten) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (ljusten) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (ljusten) [reply README.md:127]: Thanks!

File: README.md (hunk 2 above)
Comment (ljusten) [reply README.md]: Done.

**Review by chrschng — APPROVED** [review]
Thanks!

**Pull request merged**